library/utility/support/ut_enum.e

```diff
@@ -0,0 +1,43 @@
+note
+
+        description:
+
+                "Enumerated type skeleton"
+
+        library: "Gobo Eiffel Utility Library"
+        copyright: "Copyright (c) 2015, Eric Bezault and others"
+        license: "MIT License"
+
+deferred class UT_ENUM[G]
+
+inherit
+
+        ANY -- Needed for SE 2.1b1.
+
+feature {NONE}
+	-- Initialization
+
+	make(selected: G)
+		require
+			selected_not_void: selected /= Void
+		do
+			item := selected
+	end
+
+
+feature
+	-- Access
+
+	item: G
+
+
+feature {}
+
+	members: DS_SET [G]
+		deferred
+	end
+
+invariant
+	item_is_valid: members.contains (item)
```

Comment:
I don't understand why, but if I define in my xace project

```xml
<option name="assertion" value="all"/>
```
Gobo seem to ignore my assertions...
If I replace my invariant by :
```eiffel
invariant
      item_is_valid: members.containdsfdsfsdfss (item)
```
There is no Eiffel error syntax